public/pages/workflows/new_workflow/new_workflow.tsx

```diff
@@ -28,9 +29,66 @@ import {
 } from '../../../store';
 import { enrichPresetWorkflowWithUiMetadata } from './utils';
 import { getDataSourceId } from '../../../utils';
+import { getSavedObjectsClient } from '../../../services';
+import semver from 'semver';
 
 interface NewWorkflowProps {}
 
+export const getEffectiveVersion = async (
+  dataSourceId: string | undefined
+): Promise<string> => {
+  try {
+    // Don't make the API call if no dataSourceId
+    if (dataSourceId === undefined) {
+      return '2.17.0';
+    }
+
+    const dataSource = await getSavedObjectsClient().get<DataSourceAttributes>(
+      'data-source',
+      dataSourceId
+    );
+    const version = dataSource?.attributes?.dataSourceVersion || '2.17.0';
+
+    // We use backend 2.18 for now and set it to be 2.19 for frontend logic
+    if (version === '2.17.0') {
+      return '2.17.0';
+    } else {
+      return '2.19.0';
```

Comment:
Rather than hardcode 2.19, let's leverage the current version of the plugin. We can fetch from the plugin manifest file; similar to how we fetch the version compatibility details in the manifest [here](https://github.com/opensearch-project/dashboards-flow-framework/blob/main/public/utils/utils.ts#L468), for example.